### ISSUE TITLE
upip support

### ISSUE THREAD

**Issue by petrkr** [peterhinch/micropython-nano-gui#29]
Hello,
please consider upip support for easier upload to ESP32. Because rhsell have some issues with large and lot of files... So upip will safe lots of broken keyboards and USB cables ;)

```
/micropython-nano-gui> cp -r gui /pyboard/lib/
timed out or error in transfer to remote: b''
/micropython-nano-gui> ls gui/
```

**Comment by peterhinch** [comment]
I'm puzzled: I have had no problems with `rshell` on ESP32. I don't recall anyone else experiencing this, but a forum search might yield results. I use the following args to connect to ESP32:
```bash
$ rshell -p /dev/ttyUSB0 --buffer-size=30 -a
```

I've considered putting things on PyPi but I'm not keen: I maintain a lot of repos and copying them all to PyPi would add to the maintenance burden.

**Comment by petrkr** [comment]
I see.
For now I did .tar from those files and upload by using our "deploy" script. So I figured out how to do.

Also would be nice if that library will be namespaced in subdir like

nanogui.core
nanogui.driver.....

because now it's flat along other libraries and it's messy.


For more projects on pypi would be nice to use some Travis, which will do lot of job for you. For me I have only few packages now, so I do it by hand, but if we will maintain more packages, I will try to thing about travis (or some other CI) too

anyway regards for really good project
